core/src/main/scala/org/apache/spark/shuffle/ShuffleMemoryManager.scala

```diff
@@ -138,30 +150,27 @@ class ShuffleMemoryManager protected (
 
 private[spark] object ShuffleMemoryManager {
 
-  def create(conf: SparkConf, numCores: Int): ShuffleMemoryManager = {
-    val maxMemory = ShuffleMemoryManager.getMaxMemory(conf)
+  def create(
+      conf: SparkConf,
+      memoryManager: MemoryManager,
+      numCores: Int): ShuffleMemoryManager = {
+    val maxMemory = memoryManager.maxExecutionMemory
     val pageSize = ShuffleMemoryManager.getPageSize(conf, maxMemory, numCores)
-    new ShuffleMemoryManager(maxMemory, pageSize)
+    new ShuffleMemoryManager(memoryManager, pageSize)
   }
 
+  /**
+   * Create a dummy [[ShuffleMemoryManager]] with the specified capacity and page size.
+   */
   def create(maxMemory: Long, pageSizeBytes: Long): ShuffleMemoryManager = {
-    new ShuffleMemoryManager(maxMemory, pageSizeBytes)
+    val conf = new SparkConf
+    val memoryManager = new StaticMemoryManager(conf, maxMemory, Long.MaxValue)
```

Comment:
This is a totally minor nit to be picking, but do you mind explicitly naming these parameters at their call sites?  e.g.

```
new StaticMemoryManager(conf, maxExecutionMemory = maxMemory, maxStorageMemory = Long.MaxValue)
```

I think this makes the code slightly easier to read.